app/src/main/java/com/example/georgesamuel/dubaihotels/activities/DetailActivity.java

```diff
@@ -0,0 +1,169 @@
+
+import com.bumptech.glide.Glide;
+import com.example.georgesamuel.dubaihotels.R;
+import com.example.georgesamuel.dubaihotels.util.Constants;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.MapView;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class DetailActivity extends AppCompatActivity implements OnMapReadyCallback {
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.hotelImage)
+    ImageView hotelImage;
+    @BindView(R.id.hotelName)
+    TextView hotelNameTV;
+    @BindView(R.id.hotelAddress)
+    TextView hotelAddressTV;
+    @BindView(R.id.highRate)
+    TextView highRate;
+    @BindView(R.id.lowRate)
+    TextView lowRate;
+    @BindView(R.id.mapView)
+    MapView mapView;
+    private String hotelImageUrl;
+    private String hotelName;
+    private String hotelAddress;
+    private double hotelHighRate;
+    private double hotelLowRate;
+    private double hotelLongitude;
+    private double hotelLatitude;
+    private static final String MAPVIEW_BUNDLE_KEY = "MapViewBundleKey";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_detail);
+        ButterKnife.bind(this);
+
+        getAttributes();
+        init(savedInstanceState);
+        setFields();
+
+    }
+
+    private void setFields() {
+        Glide.with(DetailActivity.this).load(hotelImageUrl).into(hotelImage);
+        hotelNameTV.setText(hotelName);
+        hotelAddressTV.setText(hotelAddress);
+        highRate.setText("High rate: " + hotelHighRate);
+        lowRate.setText("Low rate: " + hotelLowRate);
+    }
+
+    private void getAttributes() {
+        hotelImageUrl = getIntent().getStringExtra(Constants.EXTRA_HOTEL_IMAGE_URL);
+        hotelName = getIntent().getStringExtra(Constants.EXTRA_HOTEL_NAME);
+        hotelAddress = getIntent().getStringExtra(Constants.EXTRA_HOTEL_ADDRESS);
+        hotelHighRate = getIntent().getDoubleExtra(Constants.EXTRA_HOTEL_HIGH, 0.0);
+        hotelLowRate = getIntent().getDoubleExtra(Constants.EXTRA_HOTEL_LOW, 0.0);
+        hotelLongitude = getIntent().getDoubleExtra(Constants.EXTRA_HOTEL_LONGITUDE, 0.0);
+        hotelLatitude = getIntent().getDoubleExtra(Constants.EXTRA_HOTEL_LATITUDE, 0.0);
+    }
+
+    private void init(Bundle savedInstanceState) {
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setTitle(getString(R.string.details));
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+        toolbar.setNavigationOnClickListener(view -> finish());
```

Comment:
when click back it will finish activity automatic , no need for listener

Reply:
if i remove line 91 it isn't work with me "not finish activity"

Reply:
Ok, you are right